libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/market-role-group.service.ts

```diff
@@ -14,32 +14,44 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 import { Injectable } from '@angular/core';
-import { MarketRole } from '@energinet-datahub/dh/market-participant/data-access-api';
+import {
+  MarketRole,
+  MarketRoleGridArea,
+} from '@energinet-datahub/dh/market-participant/data-access-api';
+import { EicFunction } from '@energinet-datahub/dh/shared/domain';
 import { EditableMarketRoleRow } from './dh-market-participant-actor-market-roles.component';
 
 @Injectable()
 export class MarketRoleGroupService {
   readonly groupRows = (rows: EditableMarketRoleRow[]) => {
-    const marketRolesMap = rows.reduce((map, row) => {
-      map.set(row.marketRole, [...(map.get(row.marketRole) || [])]);
-      if (row.gridArea)
-        map.set(row.marketRole, [
-          ...map.get(row.marketRole),
-          {
-            id: row.gridArea,
-            meteringPointTypes: row.meteringPointTypes,
-          },
-        ]);
-      return map;
-    }, new Map());
-
     const result: MarketRole[] = [];
 
-    marketRolesMap.forEach((v, k) =>
-      result.push({ marketRole: k, gridAreas: v })
-    );
+    for (const row of rows) {
+      const gridAreaId = row.gridArea;
+
+      const gridArea: MarketRoleGridArea = {
+        id: gridAreaId ?? '',
+        meteringPointTypes: row.meteringPointTypes ?? [],
+      };
+
+      const marketRoleExsits: MarketRole | undefined = result.find(
+        (x) => x.marketRole === row.marketRole
+      );
+
+      if (marketRoleExsits) {
+        marketRoleExsits.gridAreas.push(gridArea);
+        continue;
+      }
+
+      const marketRole: MarketRole = {
+        comment: row.comment,
+        marketRole: row.marketRole ?? EicFunction.Agent,
+        gridAreas: [gridArea],
+      };
+
+      result.push(marketRole);
+    }
```

Comment:
nitpick: can see that some properties now have default values. As long as those are the correct values according to your team/domain, I don't have an opinion on it.

suggestion: consider adding a test to the related spec file that verifies the default values. 

nitpick: not related to the PR but a general comment for a future refactor. This class has nothing special in it that requires it to be an injectable service. The `groupRows` method might as well be a simple function.

Reply:
Added a test for comment https://github.com/Energinet-DataHub/greenforce-frontend/pull/1114/commits/c4156a187418040c3b11b7e9c1e5458fee8c3ae4

Not sure about testing the default values. They are only there to make typescript happy. The row validation will prevent invalid data being submitted.